### PR TITLE
fix(embedding): remove forced /v1 suffix for embeddings

### LINKED ISSUE #6558: [Feature]能否把添加部分模型提供商时API Base URL尾部自动补上 /v1的功能删去
### Description / 描述

Image

能否把添加部分模型提供商时API Base URL尾部自动补上 /v1/...的功能删去，会导致无法接入部分模型，如glm等

Image

在删去尾部的v4后仍无法接入

Image

Image

### Use Case / 使用场景

添加模型提供商的bug（？）

### Willing to Submit PR? / 是否愿意提交PR？

- [ ] Yes, I am willing to submit a PR. / 是的，我愿意提交 PR。

### Code of Conduct

- [x] I have read and agree to abide by the project's [Code of Conduct](https://docs.github.com/zh/site-policy/github-terms/github-community-code-of-conduct). /

### PR DESCRIPTION
**Pull request by ccsang**
Fixes #6558

## Problem
AstrBot's embedding adapter unconditionally appends /v1 to all API base URLs.

## Solution
Add embedding_api_base_mode config option (auto|force_v1|as_is).

## Testing
- ✅ All 7 new tests pass

Co-Authored-By: Claude Sonnet 4.6 <noreply@anthropic.com>

## Summary by Sourcery

Make OpenAI-compatible embedding providers use a shared helper to resolve the API base URL and support configurable handling of the /v1 suffix.

New Features:
- Add a configurable embedding_api_base_mode option (auto|force_v1|as_is) to control how /v1 is appended to embedding API base URLs.

Bug Fixes:
- Prevent embedding API base URLs from being incorrectly forced to end with /v1 when a different path is required.

Enhancements:
- Introduce a reusable resolve_openai_compatible_base_url utility for consistent base URL handling across providers.

Tests:
- Add unit and async tests covering resolve_openai_compatible_base_url behavior and embedding provider base URL resolution for all modes and edge cases.

### CODE REVIEW

File: astrbot/core/provider/utils.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+    Returns:
+        The resolved API base URL.
+    """
+    api_base = api_base.strip()
+    if not api_base:
+        return default_base
+
+    if mode == "as_is":
+        return api_base.removesuffix("/")
```
[the remark this answers is not in the document]
Comment (ccsang) [reply astrbot/core/provider/utils.py]: ✅ Fixed in commit cad7d069. The `as_is` mode now returns `api_base` unchanged without removing trailing slashes. Docstring updated to clarify this behavior.

File: astrbot/core/config/default.py [hunk 2]
```diff
@@ -1937,6 +1938,11 @@ class ChatProviderTemplate(TypedDict):
                         "description": "API Base URL",
                         "type": "string",
                     },
+                    "embedding_api_base_mode": {
+                        "description": "API Base URL Mode",
+                        "type": "string",
+                        "hint": "控制 API Base URL 的 /v1 后缀处理方式。auto: 自动判断（默认），如果 URL 不以 /v1 结尾则添加。force_v1: 强制添加 /v1 后缀。as_is: 保持原样不做修改。",
+                    },
```
[the remark this answers is not in the document]
Comment (ccsang) [reply astrbot/core/config/default.py:1946]: ✅ Fixed in commit cad7d069. Added enum constraint `["auto", "force_v1", "as_is"]` to prevent misconfiguration.

File: astrbot/core/provider/utils.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        api_base = api_base.removesuffix("/")
+        if not api_base.endswith("/v1"):
+            api_base = f"{api_base}/v1"
+        return api_base
+
+    # mode == "auto"
+    api_base = api_base.removesuffix("/")
+    if not api_base.endswith("/v1"):
+        api_base = f"{api_base}/v1"
+    return api_base
```
[the remark this answers is not in the document]
Comment (ccsang) [reply astrbot/core/provider/utils.py:38]: ✅ Fixed in commit daf0fcc9. Simplified code by merging identical `force_v1` and `auto` logic into a single code path.

File: tests/test_openai_embedding_source.py [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+
+@pytest.mark.asyncio
+async def test_openai_embedding_provider_with_trailing_slash():
+    """Test OpenAI Embedding provider removes trailing slash in as_is mode."""
+    provider = _make_provider(
+        {"embedding_api_base": "https://api.example.com/", "embedding_api_base_mode": "as_is"}
+    )
+    try:
+        # The provider removes trailing slash, but OpenAI client adds it back
+        assert str(provider.client.base_url) == "https://api.example.com"
```
[the remark this answers is not in the document]
Comment (ccsang) [reply tests/test_openai_embedding_source.py]: ✅ Fixed in commit daf0fcc9. Updated test assertion to expect trailing slash from OpenAI client, which is the actual behavior.

[the remark this answers is not in the document]
Comment (ccsang) [reply tests/test_openai_embedding_source.py]: ✅ Fixed in commit daf0fcc9. Updated test assertion to expect `https://api.example.com/` with trailing slash, matching the actual OpenAI client behavior.

File: astrbot/core/provider/utils.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (ccsang) [reply astrbot/core/provider/utils.py]: ✅ Addressed in commit daf0fcc9. Added documentation in both the function docstring and config hint to clarify that `as_is` mode should only provide base URL without `/embeddings` path, as the OpenAI SDK will append it automatically.

**Comment by Soulter** [comment]
#6669 fixed
